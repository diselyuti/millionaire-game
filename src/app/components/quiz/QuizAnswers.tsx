import React from 'react';
import QuizAnswerInactive from '@/app/components/quiz/QuizAnswerInactive.tsx';
import styles from '@/app/styles/components/quiz/quiz-answers.module.scss';
import { IQuizAnswer, QuizAnswerStatus } from '@/types/index.ts';
import QuizAnswerWrong from '@/app/components/quiz/QuizAnswerWrong.tsx';
import QuizAnswerSelected from '@/app/components/quiz/QuizAnswerSelected.tsx';
import QuizAnswerCorrect from '@/app/components/quiz/QuizAnswerCorrect.tsx';
import { useDispatch } from 'react-redux';
import { selectAnswer } from '@/stores/progressSlice.ts';

function QuizAnswers({ answers }: { answers: IQuizAnswer[] }) {
  const dispatch = useDispatch();
  const handleAnswerClick = (answer: IQuizAnswer) => {
    dispatch(selectAnswer(answer));
  };

  const charCodeA = 'A'.charCodeAt(0);

  return (
    <section className={styles.answers}>
      {answers.map((answer, index) => {
        const letter = String.fromCharCode(charCodeA + index);

        if (answer.status === QuizAnswerStatus.Correct) {
          return (
            <QuizAnswerCorrect
              key={answer.id}
              letter={letter}
              text={answer.answer}
            />
          );
        }
        if (answer.status === QuizAnswerStatus.Wrong) {
          return (
            <QuizAnswerWrong
              key={answer.id}
              letter={letter}
              text={answer.answer}
            />
          );
        }

        if (answer.status === QuizAnswerStatus.Selected) {
          return (
            <QuizAnswerSelected
              key={answer.id}
              letter={letter}
              text={answer.answer}
            />
          );
        }

        return (
          <QuizAnswerInactive
            key={answer.id}
            letter={letter}
            text={answer.answer}
            onClick={() => handleAnswerClick(answer)}
          />
        );
      })}
    </section>
  );
}

export default QuizAnswers;
