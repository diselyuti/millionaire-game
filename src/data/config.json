{
  "currentQuestion": 0,
  "steps": [
    {
      "id": 1,
      "prize": 100,
      "selected": false,
      "finished": false
    },
    {
      "id": 2,
      "prize": 200,
      "selected": false,
      "finished": false
    },
    {
      "id": 3,
      "prize": 300,
      "selected": false,
      "finished": false
    },
    {
      "id": 4,
      "prize": 500,
      "selected": false,
      "finished": false
    },
    {
      "id": 5,
      "prize": 1000,
      "selected": false,
      "finished": false
    },
    {
      "id": 6,
      "prize": 2000,
      "selected": false,
      "finished": false
    },
    {
      "id": 7,
      "prize": 4000,
      "selected": false,
      "finished": false
    },
    {
      "id": 8,
      "prize": 8000,
      "selected": false,
      "finished": false
    }
  ],
  "isGameOver": false,
  "questions": [
    {
      "question": "Якими кольорами складається прапор України?",
      "correctAnswerId": 1,
      "prizeId": 1,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Жовтий та синій"
        },
        {
          "id": 2,
          "answer": "Червоний та чорний"
        },
        {
          "id": 3,
          "answer": "Зелений та блакитний"
        },
        {
          "id": 4,
          "answer": "Червоний та білий"
        }
      ]
    },
    {
      "question": "Який океан є найбільшим за площею?",
      "correctAnswerId": 2,
      "prizeId": 2,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Атлантичний океан"
        },
        {
          "id": 2,
          "answer": "Тихий океан"
        },
        {
          "id": 3,
          "answer": "Індійський океан"
        },
        {
          "id": 4,
          "answer": "Північний Льодовитий океан"
        }
      ]
    },
    {
      "question": "Хто написав роман \"Майстер і Маргарита\"?",
      "correctAnswerId": 3,
      "prizeId": 3,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Лев Толстой"
        },
        {
          "id": 2,
          "answer": "Федор Достоєвський"
        },
        {
          "id": 3,
          "answer": "Михайло Булгаков"
        },
        {
          "id": 4,
          "answer": "Антон Чехов"
        }
      ]
    },
    {
      "question": "Який материк має найбільшу площу?",
      "correctAnswerId": 3,
      "prizeId": 4,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Африка"
        },
        {
          "id": 2,
          "answer": "Австралія"
        },
        {
          "id": 3,
          "answer": "Євразія"
        },
        {
          "id": 4,
          "answer": "Північна Америка"
        }
      ]
    },
    {
      "question": "Як називається столиця Японії?",
      "correctAnswerId": 3,
      "prizeId": 5,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Киото"
        },
        {
          "id": 2,
          "answer": "Осака"
        },
        {
          "id": 3,
          "answer": "Токіо"
        },
        {
          "id": 4,
          "answer": "Хоккайдо"
        }
      ]
    },
    {
      "question": "Хто зображений на найпоширеніших банкнотах долара США?",
      "correctAnswerId": 2,
      "prizeId": 6,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Джордж Вашингтон"
        },
        {
          "id": 2,
          "answer": "Авраам Лінкольн"
        },
        {
          "id": 3,
          "answer": "Бенджамін Франклін"
        },
        {
          "id": 4,
          "answer": "Томас Джефферсон"
        }
      ]
    },
    {
      "question": "Як називається найвища гора світу?",
      "correctAnswerId": 2,
      "prizeId": 7,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Монблан"
        },
        {
          "id": 2,
          "answer": "Еверест"
        },
        {
          "id": 3,
          "answer": "Кіліманджаро"
        },
        {
          "id": 4,
          "answer": "Аконкагуа"
        }
      ]
    },
    {
      "question": "Як називається фігура з чотирма однаковими сторонами і чотирма прямими кутами?",
      "correctAnswerId": 4,
      "prizeId": 8,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Круг"
        },
        {
          "id": 2,
          "answer": "Трикутник"
        },
        {
          "id": 3,
          "answer": "Прямокутник"
        },
        {
          "id": 4,
          "answer": "Квадрат"
        }
      ]
    }
  ]
}
