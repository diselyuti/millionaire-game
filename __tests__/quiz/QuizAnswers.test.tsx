import React from 'react';
import '@testing-library/jest-dom';
import QuizAnswers from "@/app/components/quiz/QuizAnswers.tsx";
import {render} from "@testing-library/react";
import config from "@/data/config.json";

jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));

describe('QuizAnswers', () => {
  test('should render texts', () => {
    const { container } = render(<QuizAnswers answers={config.questions[0].answers}/>);
    expect(container).toHaveTextContent('A');
    expect(container).toHaveTextContent('B');
    expect(container).toHaveTextContent('C');
    expect(container).toHaveTextContent('D');
    expect(container).toHaveTextContent('Жовтий та синій');
    expect(container).toHaveTextContent('Червоний та чорний');
    expect(container).toHaveTextContent('Зелений та блакитний');
    expect(container).toHaveTextContent('Червоний та білий');
  });

  test('should render QuizAnswers', () => {
    const { container } = render(<QuizAnswers answers={config.questions[0].answers}/>);
    expect(container).toMatchSnapshot();
  });
});
