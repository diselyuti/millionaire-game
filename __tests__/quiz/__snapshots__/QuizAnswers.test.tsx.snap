// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizAnswers should render QuizAnswers 1`] = `
<div>
  <section
    class="answers"
  >
    <button
      class="hexagon"
      type="button"
    >
      <p
        class="answerText"
      >
        <span
          class="letter"
        >
          A
        </span>
        <span>
          Жовтий та синій
        </span>
      </p>
    </button>
    <button
      class="hexagon"
      type="button"
    >
      <p
        class="answerText"
      >
        <span
          class="letter"
        >
          B
        </span>
        <span>
          Червоний та чорний
        </span>
      </p>
    </button>
    <button
      class="hexagon"
      type="button"
    >
      <p
        class="answerText"
      >
        <span
          class="letter"
        >
          C
        </span>
        <span>
          Зелений та блакитний
        </span>
      </p>
    </button>
    <button
      class="hexagon"
      type="button"
    >
      <p
        class="answerText"
      >
        <span
          class="letter"
        >
          D
        </span>
        <span>
          Червоний та білий
        </span>
      </p>
    </button>
  </section>
</div>
`;
